icicle/include/icicle/program/program.h

```diff
@@ -28,70 +28,93 @@ namespace icicle {
   class Program
   {
   public:
-    // Generate a program based on a lambda function
-    Program(std::function<Symbol<S>(std::vector<Symbol<S>>&)> program_func, const int nof_inputs)
+    // Generate a program based on a lambda function with multiple inputs and multiple outputs
+    Program(std::function<void(std::vector<Symbol<S>>&)> program_func, const int nof_parameters)
+        : m_nof_parameters(nof_parameters)
     {
-      std::vector<Symbol<S>> program_inputs(nof_inputs);
-      set_as_inputs(program_inputs);
-      Symbol<S> result = program_func(program_inputs);
-      generate_program(result);
+      // Generate the DFG
+      std::vector<Symbol<S>> program_parameters(m_nof_parameters);
+      set_as_inputs(program_parameters);
```

Comment:
program parameters include outputs right? then why do you set all as input?

Reply:
because its potentially an input.
After the lambda runs, the one that changed will contains an operation that != INPUT